cardea/featurization/featurization.py

```diff
@@ -6,53 +6,95 @@ class Featurization():
-        return ["day", "month", "year", "weekday", "is_weekend"]
+    TRANS_PRIMITIVES = ["day", "month", "year", "weekday", "is_weekend"]
 
-    @staticmethod
-    def n_jobs():
-        return 1
-
-    @staticmethod
-    def max_depth():
-        return 2
-
-    def generate_feature_matrix(self, es, target, cutoff, verbose=True):
+    def generate_feature_matrix(self, es, target, label_times,
+                                instance_ids=None, agg_primitives=AGG_PRIMITIVES,
+                                trans_primitives=TRANS_PRIMITIVES, max_depth=2,
+                                ignore_entities=None, ignore_variables=None, seed_features=None,
+                                drop_contains=None, drop_exact=None, max_features=-1,
+                                training_window=None, n_jobs=1, verbose=False,
+                                include_cutoff_time=True, encode=False):
         """Calculates a feature matrix and features given in Featurization object.
 
-          Args:
+        Args:
             es (featuretools.EntitySet):
-              An already initialized entityset. Required if entities and relationships
-              are not defined.
+                An already initialized entityset.
             target (str):
-              A string of the target entity name.
-            cutoff (pandas.DataFrame):
-              Specified times at which to calculate the features for each instance.
+                Entity id of entity on which to make predictions.
+            label_times (pandas.DataFrame):
+                Specifies times at which to calculate the features for each instance.
+                The resulting feature matrix will use data up to and including the cutoff time.
+                The instance ids for which to calculate features must be in a column with the
+                same name as the target entity index or a column named ``instance_id``.
+                The cutoff time values in the DataFrame must be in a column with the same name
+                as the target entity time index or a column named ``time``.
+                If the DataFrame has more than two columns, any additional columns will be added
+                to the resulting feature matrix.
+            instance_ids (list):
+                List of instances on which to calculate features.
+            agg_primitives (list):
+                List of Aggregation Feature types to apply.
+            trans_primitives (list):
+                List of Transform Feature functions to apply.
+            max_depth (int):
+                Maximum allowed depth of features.
+            ignore_entities (list):
+                List of entities to blacklist when creating features.
+            ignore_variables (dict):
+                List of specific variables within each entity to blacklist when creating features.
+            seed_features (list):
+                List of manually defined features to use.
+            drop_contains (list):
```

Comment:
what does the name here refer to? If the name is a feature name, then how you generate feature names?

Reply:
It can pertain to either the name of the column or name of the operation. For example, say you have column `hypertension`, and you don't want any features that use that column, you can use this parameter to highlight this constraint.